ast/compile.go

```diff
@@ -660,6 +674,19 @@ func (c *Compiler) checkTypes() {
 	c.TypeEnv = env
 }
 
+func (c *Compiler) checkUnsafeBuiltins() {
+	for _, name := range c.sorted {
+		mod := c.Modules[name]
+		WalkRules(mod, func(r *Rule) bool {
```

Comment:
Nit: see comment below about small refactoring to `checkUnsafeBuiltinsInBody`. If that updates then we don't have to call WalkRules here.